drivers/counter/counter_nrfx_timer.c

```diff
@@ -90,8 +90,15 @@ static int counter_nrfx_set_alarm(struct device *dev, u8_t chan_id,
 		return -EBUSY;
 	}
 
-	cc_val = alarm_cfg->ticks + (alarm_cfg->absolute ?
-				0 : nrfx_timer_capture(timer, COUNTER_READ_CC));
+	if (alarm_cfg->absolute) {
+		cc_val = alarm_cfg->ticks;
+	} else {
+		cc_val = alarm_cfg->ticks +
+				nrfx_timer_capture(timer, COUNTER_READ_CC);
+		cc_val -= (cc_val > counter_get_top_value(dev)) ?
+				counter_get_top_value(dev) : 0;
```

Comment:
I am afraid it still may fail to set the proper compare value. Consider the following (contrived but still) scenario:
- top value is `0xFFFFFF00`
- `nrfx_timer_capture(timer, COUNTER_READ_CC)` returns `0xFFFFF100`
- an alarm with relative ticks of `0x1000` is requested

`cc_val` will be set to `0x100` then, while it should be `0x200`.

Maybe something like the code below could work?
```
u32_t remainder;
...
cc_val = nrfx_timer_capture(timer, COUNTER_READ_CC);
remainder = (counter_get_top_value(dev) - cc_val);
if (remainder > alarm_cfg->ticks) {
    cc_val += alarm_cfg->ticks;
} else {
    cc_val = alarm_cfg->ticks - remainder;
}
```

Reply:
good catch. I've refactored the algorithm in `counter_nrfx_timer.c` to separate function.

Reply:
> I've refactored the algorithm in `counter_nrfx_timer.c` to separate function.

Good. Shouldn't the RTC based implementation use a similar solution?

Reply:
it's 24bit so there is no risk so i left it but if you insist. I felt that first argument is more straight forward (but was failing due to 32 bit overflow).

Reply:
Oh, right, the top value is 24-bit and the relative ticks value cannot be bigger than it. So there won't be a 32-bit overflow when these two are added. Maybe a short comment just stating that there is no risk of the overflow would be a good thing then?

Reply:
done